### DISCUSSION TITLE
How to apply sorting on HxGrid with EF Core

### DISCUSSION THREAD

**Discussion by cvboucher** [havit/Havit.Blazor#239] [Q&A] [answered]
What's the best way to apply server side sorting to an EF Core IQueryable from an HXGrid?

I tried setting the SortKeySelector on the HxGridColumn:

`SortKeySelector="@(item => item.PrincipalName)"`

Then applying it to the IQueryable in a generic helper method I wrote:

`query = query.OrderBy(sort.SortKeySelector);`

This works for string/varchar columns, but returns an IComparable for other data types which causes EF to throw an exception.

I see in the documentation that SortKeySelector is for client side sorting.  Is there a way to do server side sorting with EF Core?

Thanks,
Craig

**Comment by hakenr** [comment, marked as answer]
Please see the discussion and suggestions in this issue: https://github.com/havit/Havit.Blazor/issues/91
